File: backend/app/models/label.py
from sqlalchemy import String, ForeignKey, Boolean, BigInteger
from sqlalchemy.orm import Mapped, mapped_column, relationship
from .base import Base, AuditMixin, SoftDeleteMixin

class Label(Base, AuditMixin, SoftDeleteMixin):
    __tablename__ = "labels"
    
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    color: Mapped[str] = mapped_column(String(50), default="#6B7280")
    description: Mapped[str | None] = mapped_column(String(255), nullable=True)
    
    project_id: Mapped[int] = mapped_column(BigInteger, ForeignKey("projects.id"), nullable=False)
    
    project: Mapped["Project"] = relationship(back_populates="labels")
    issue_links: Mapped[list["IssueLabel"]] = relationship(back_populates="label")

File: backend/internal/model/label.go
package model

// Label represents a project label for categorizing issues.
type Label struct {
	BaseModel

	Name        string  `gorm:"size:255;not null" json:"name"`
	Color       string  `gorm:"size:50;default:#6B7280" json:"color"`
	Description *string `gorm:"size:255" json:"description"`
	ProjectID   *uint64 `gorm:"index" json:"project_id"`
	WorkspaceID uint64  `gorm:"not null" json:"workspace_id"`

	// Relationships
	Project    Project      `gorm:"foreignKey:ProjectID" json:"-"`
	IssueLinks []IssueLabel `gorm:"foreignKey:LabelID" json:"-"`
}

func (Label) TableName() string {
	return "labels"
}
